### PR TITLE
Ds 17734

### PR DESCRIPTION
**Pull request by karishma-pingidentity**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,12 +15,8 @@ Data Governance Broker sample UI for delegated account management
    group, you can apply the script to the entire server group using the "--applyChangeTo server-group" argument. For
    Data Governance Brokers added later, untar the account-manager archive before cloning the configuration from an
    existing server.
-3. Optionally modify the OAuth2 Scope Policy using oauth2-scope-policy.xml to allow users with an entitlement matching a
```
Comment (chad-ubid) [inline README.md]: Unfortunately it won't let me leave a comment on a line in the file not included in the commit info.  Line #13 of this file mentions adding Scope and OAuth Client objects - it should no longer include OAuth Client objects.

File: README.md [hunk 2]
```diff
@@ -15,12 +15,8 @@ Data Governance Broker sample UI for delegated account management
    group, you can apply the script to the entire server group using the "--applyChangeTo server-group" argument. For
```
Comment (chad-ubid) [inline README.md]: Unfortunately it won't let me leave a comment on a line in the file not included in the commit info so I am leaving it here...

Line #13 of this file mentions adding several things including OAuth Client objects - it should no longer include OAuth Client objects.

File: README.md [hunk 3]
```diff
@@ -50,11 +46,7 @@ dsconfig create-http-servlet-cross-origin-policy --policy-name account-manager \
```
Comment (chad-ubid) [inline README.md]: Same story with line 31 above - not OAuth Client.

Also, step 6 should be removed.

File: README.md [hunk 4]
```diff
@@ -50,11 +46,7 @@ dsconfig create-http-servlet-cross-origin-policy --policy-name account-manager \
...
 
-8. Optionally modify the OAuth2 Scope Policy using oauth2-scope-policy.xml to allow users with an entitlement matching a
-   tag on a resource scope to have access to that scope (setup.dsconfig contains a commented-out command for modifying
-   this policy).  The scopes installed by setup.dsconfig have "csr" and "csr-limited" tags.  Add entitlements matching
-   the tags to user accounts for use with this sample (see the Entitlements section below).
-9. Access the sample at your servlet container's address and the appropriate context.
+8. Access the sample at your servlet container's address and the appropriate context.
 
 
 ### Deployment with PingFederate as the Identity Provider
```
Comment (chad-ubid) [inline README.md]: This section needs to be updated.  It still says the default id for Broker to be IDP.

The first paragraph should say something like "The application's default configuration assumes a Data Governance Broker server is performing the Resource Server role and a PingFederate server is performing the Identity Provider (IDP) role."

The rest should be OK.

File: app/index/index.html.ts [hunk 5]
```diff
@@ -64,25 +64,6 @@ export const template = `
     <li [routerLinkActive]="['active']"
```
Comment (chad-ubid) [inline app/index/index.html.ts]: Line 50 - 52 above should be removed (reset password action)

File: app/index/index.html.ts [hunk 6]
```diff
@@ -64,25 +64,6 @@ export const template = `
...
-        role="presentation"><a [routerLink]="['/session']" 
-                               href="javascript:void(0)">Sessions</a></li>
-    <li *ubidIfFunctionality="functionalityEnum.Consents"
-        [routerLinkActive]="['active']"
-        role="presentation"><a [routerLink]="['/consent']"
-                               href="javascript:void(0)">Consent</a></li>
-    <li [routerLinkActive]="['active']"
-        role="presentation"><a [routerLink]="['/preference']"
-                               href="javascript:void(0)">Preferences</a></li>
   </ul>
```
Comment (chad-ubid) [inline app/index/index.html.ts]: The 3 lines above this one should not have been removed.  Preferences is still a valid link

File: app/shared/services/scim.service.ts [hunk 7]
```diff
@@ -212,21 +137,13 @@ export class ScimService {
...
-      this.consents.next(undefined);
-      this.externalIdentities.next(undefined);
-      this.validatedEmailAddress.next(undefined);
-      this.validatedPhoneNumber.next(undefined);
-      this.totpSharedSecret.next(undefined);
     }
     if (profile) {
       this.profile.next(profile);
       if (clearCache) {
         // also, pre-fetch the account state and password quality requirements so we have them up front if they disable
```
Comment (chad-ubid) [inline app/shared/services/scim.service.ts]: password quality requirements should be removed from this comment

File: app/shared/services/scim.service.ts [hunk 8]
```diff
@@ -389,9 +188,7 @@ export class ScimService {
     var query = 'userName ' + nonIndexedOperation + ' "' + filter + '" or ' +
         'name.givenName ' + indexedOperation + ' "' + filter + '" or ' +
         'name.familyName ' + indexedOperation + ' "' + filter + '" or ' +
-        'name.formatted ' + indexedOperation + ' "' + filter + '" or ' +
-        'emails.value ' + nonIndexedOperation + ' "' + filter + '" or ' +
-        'phoneNumbers.value ' + nonIndexedOperation + ' "' + filter + '"';
```
Comment (chad-ubid) [inline app/shared/services/scim.service.ts]: These lines should be added back in - this lets them search by email or phonenumber

File: assets/css/ubid-account.css [hunk 9]
```diff
@@ -686,7 +612,7 @@ h5 {
   word-break: break-all;
 }
 .break-word-or-hyphen {
-  word-break: break-word; /* non-standard for webkit */
+  word-break: break-all; /* non-standard for webkit */
```
Comment (chad-ubid) [inline assets/css/ubid-account.css]: Was this change intentional?

File: setup.dsconfig [hunk 10]
```diff
@@ -4,40 +4,17 @@
 # Handler objects for the Data Governance Broker to host it.
```
Comment (chad-ubid) [inline setup.dsconfig]: This comment should be changed to something like ```This dsconfig batch file creates the OAuth2 Scope objects within the Data Governance Broker and configures the Web Application Extension and Connection Handler objects for the Data Governance Broker to host the Account Manager Sample.```

File: setup.dsconfig [hunk 11]
```diff
@@ -92,22 +66,9 @@ dsconfig create-oauth2-scope \
...
-    --set tag:csr \
-    --set tag:csr-limited
-
-dsconfig create-oauth2-scope \
     --scope-name urn:pingidentity:scope:admin:change_password \
     --type resource \
     --set "description:Change and reset passwords." \
-    --set "consent-prompt-text:Change and reset passwords." \
     --set scim-resource-type:Users \
     --set "scim-sub-resource-type:Password Update" \
```
Comment (chad-ubid) [inline setup.dsconfig]: Does this dsconfig command work?  I thought the "Password Update" scim sub resource type was removed?  If it is broken, I think that line could be removed and resource-attribute could be set to "password" instead of "*"

File: setup.dsconfig [hunk 12]
```diff
@@ -92,22 +66,9 @@ dsconfig create-oauth2-scope \
     --set tag:csr
```
Comment (chad-ubid) [inline setup.dsconfig]: All of the "--set tag:" lines can be removed from this file (csr, csr-limited, etc.)

File: README.md [hunk 13]
```diff
@@ -276,26 +203,17 @@ can be found near the top of the script (search for the "export" statements). Va
    The redirect URI for the client in the OAuth flow.  This should be the address used to view the sample, and
    should be one of the Redirect URLs configured for the sample OAuth2 Client in the Data Governance Broker.  A value
    like "https://1.2.3.4:8443/samples/account-manager/" should be used.
-4. `IDENTITY_PROVIDER_TYPE`
-   The type of IDP referenced by `IDENTITY_PROVIDER_URL`.  Should be set to `IdentityProviderTypes.Broker` or
-   `IdentityProviderTypes.PingFederate`.
-5. `CLIENT_ID`
+4. `CLIENT_ID`
    The Client ID assigned to the Account Manager OAuth2 Client in the Data Governance Broker configuration.  This is set
```
Comment (karishma-pingidentity) [inline README.md]: Should I be doing this change for steps 5 and 6 as well?

File: assets/css/ubid-account.css (hunk 9 above)
Comment (karishma-pingidentity) [inline assets/css/ubid-account.css]: It was showing up as an error to have "break-word" and when I hovered over it, it seemed "break-all" was the correct syntax. However, I can revert it. The revert still compiles.

File: README.md [hunk 14]
```diff
@@ -10,17 +10,13 @@ Data Governance Broker sample UI for delegated account management
 
 1. Extract the account-manager.tar.gz file (found in the samples directory).
 2. Use dsconfig to run the commands in the setup.dsconfig file.  This will create a Web Application Extension, assign
-   the Web Application Extension to the HTTPS Connection Handler, and add the required Scope and OAuth Client objects to
+   the Web Application Extension to the HTTPS Connection Handler, and add the required Scope to
```
Comment (chad-ubid) [inline README.md]: "Scope" should be "Scopes" now that "objects" was removed from the sentence.

File: README.md [hunk 15]
```diff
@@ -32,11 +28,9 @@ Data Governance Broker sample UI for delegated account management
    (see the "Customization" section for additional details).
 4. Deploy the custom account-manager.war file into your servlet container as appropriate (e.g. copy it into the webapps
    directory of your Tomcat installation and restart).
-5. Use dsconfig to run the commands in the setup.dsconfig file that add the required Scope and OAuth Client objects to
+5. Use dsconfig to run the commands in the setup.dsconfig file that add the required Scope to
```
Comment (chad-ubid) [inline README.md]: "Scope" should be "Scopes" now that "objects" was removed from the sentence.

File: app/shared/services/scim.service.ts [hunk 16]
```diff
@@ -141,7 +141,7 @@ export class ScimService {
     if (profile) {
       this.profile.next(profile);
       if (clearCache) {
-        // also, pre-fetch the account state and password quality requirements so we have them up front if they disable
+        // also, pre-fetch the account state so we have it up front if it disables
```
Comment (chad-ubid) [inline app/shared/services/scim.service.ts]: This comment should now be `also, pre-fetch the account state so we have it up front if they disable the account`

File: setup.dsconfig [hunk 17]
```diff
@@ -62,20 +57,16 @@ dsconfig create-oauth2-scope \
...
 
 dsconfig create-oauth2-scope \
     --scope-name urn:pingidentity:scope:admin:change_password \
     --type resource \
     --set "description:Change and reset passwords." \
     --set scim-resource-type:Users \
-    --set "scim-sub-resource-type:Password Update" \
-    --set "resource-attribute:*" \
+    --set "resource-attribute:password" \
     --set resource-operation:retrieve \
```
Comment (chad-ubid) [inline setup.dsconfig]: remove this - don't want to allow password retrieve

File: setup.dsconfig [hunk 18]
```diff
@@ -25,8 +22,6 @@ dsconfig create-oauth2-scope \
     --set resource-operation:modify \
     --set resource-operation:retrieve \
     --set resource-operation:search \
```
Comment (chad-ubid) [inline setup.dsconfig]: remove line continuation on last line `\`

File: app/shared/services/scim.service.ts [hunk 19]
```diff
@@ -141,8 +141,7 @@ export class ScimService {
     if (profile) {
       this.profile.next(profile);
       if (clearCache) {
-        // also, pre-fetch the account state so we have it up front if it disables
-        // account or change password
+        // also, pre-fetch the account state so we have it up front if they disable the account or change password
```
Comment (chad-ubid) [inline app/shared/services/scim.service.ts]: remove "or change password"

File: app/app.config.ts [hunk 20]
```diff
@@ -8,73 +8,32 @@ import { Http, Response } from '@angular/http';
...
-  LOGOUT_ENDPOINT: 'idp/startSLO.ping'
-};
 
+export const IDENTITY_PROVIDER_AUTH_ENDPOINT = 'as/authorization.oauth2';
+export const IDENTITY_PROVIDER_LOGOUT_ENDPOINT = '/idp/startSLO.ping';
 
-export const CLIENT_ID = '@account-manager@';
 
+export const CLIENT_ID = 'test';
+//export const CLIENT_ID = '@account-manager@';
```
Comment (chad-ubid) [inline app/app.config.ts]: this still needs to be updated
